config/routes.rb

```diff
@@ -1,10 +1,34 @@
 Rails.application.routes.draw do
+
+
+  devise_for :users, :controllers => { omniauth_callbacks: "users/omniauth_callbacks", registrations: 'users/registrations' }
+
+
+
+  root to: 'visitors#index'
+  get '/playlists/:playlist_id/songs/search', to: 'songs#search'
+
+  get '/playlists/find', to: 'playlists#find'
```

Comment:
You can create these same routes within the `resources` block [using member and collection](http://guides.rubyonrails.org/routing.html#adding-more-restful-actions).

```ruby
resources :playlists do
  get 'find', to: 'playlists#find', as: :collection # /playlists/find

  resources :songs do
    get 'search', to: 'songs#search', as: :collection  # /playlists/:id/songs/search
  end
end
```